edited multipath text for clarity (tech writer) (#155)
canonical/ubuntu-server-documentation: explanation/intro-to/multipath.md

(introduction-to-multipath)=
# Introduction to device mapper multipathing ("multipath")

> Device mapper multipathing (which will be called **multipath** in this document) allows you to create a virtual device that aggregrates multiple input/output (I/O) paths between server nodes and storage arrays. These I/O paths are physical storage area network (SAN) connections that can include separate cables, switches, and controllers.

Author: :)

| `multipathd` daemon | Monitors paths; as paths fail and come back, it may initiate path group switches. Provides for interactive changes to multipath devices. This daemon must be restarted for any changes to the `/etc/multipath.conf` file to take effect. |
| `kpartx` command | Creates device-mapper devices for the partitions on a device. It is necessary to use this command for DOS-based partitions with multipath. The `kpartx` is provided in its own package, but the `multipath-tools` package depends on it.  |

## Multipath setup overview

The multipath setup process is usually simple because it has compiled-in default settings that are suitable for common multipath configurations. The basic procedure for configuring your system with multipath is:

1.  Install the `multipath-tools` and `multipath-tools-boot` packages.

2.  Create an empty config file called `/etc/multipath.conf`.

3.  Edit the `multipath.conf` file to modify default values and save the updated file.

4.  Start the multipath daemon.

> 5.  Update initial RAM disk.

Reviewer: Not a blocker, but just FYI - in markdown you don't need to number the list sequentially. Sphinx is smart enough to handle the numbering even if all the list entries are numbered with `1`, which is very convenient if you're moving things around.

For detailed setup instructions for multipath configuration see {ref}`DM-Multipath configuration <configuring-multipath>` and {ref}`DM-Multipath setup <multipath-configuration-examples>`.

## Example of a multipath device in use

> For example, a node with two host bus adapters (HBAs) attached to a storage controller, with two ports, via a single un-zoned {term}`FC` switch sees four devices:  `/dev/sda`, `/dev/sdb`, `/dev/sdc`, and `/dev/sdd`. If you use multipath with this node, a single multipath device is created with a unique WWID that reroutes I/O to those four underlying devices according to the multipath configuration.

Reviewer: I prefer the original text of the second sentence, since it's written in a more active voice ("Multipath creates a single device"). This change uses the passive voice ("a single multipath device is created"), which we try to avoid wherever possible. The active voice is usually more straightforward to parse for readers who have English as a second (or +) language.

If you try to create an LVM physical volume on a whole device on which you have configured partitions, the `pvcreate` command will fail.
```

Once you create an LVM logical volume that has active/passive multipath arrays as the underlying physical devices, you must add filters in the `lvm.conf` file to exclude the disks that underlie the multipath devices. This is because if the array automatically changes the active path to the passive path when it receives I/O, multipath will failover and {term}`fallbacks` whenever LVM scans the passive path if these devices are not filtered.

> If an active/passive arrays requires a command to make the passive path active, LVM will print a warning message when this occurs. To filter all SCSI devices in the LVM configuration file (`lvm.conf`), include the following filter in the devices section of the file:

Reviewer: ```suggestion
If an active/passive array requires a command to activate the passive path, LVM will print a warning message. To filter all SCSI devices in the LVM configuration file (`lvm.conf`), include the following filter in the devices section of the file:
```

Reviewer: I think this is a little bit tighter :)
